controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -197,6 +199,16 @@ def test_delete_volume_still_copy_fcmaps_not_removed(self):
         with self.assertRaises(array_errors.ObjectIsStillInUseError):
             self.svc.delete_volume("volume")
 
+    def test_delete_volume_found_hyper_swap(self):
+        fcmaps_as_target = Mock(as_list=[])
```

Comment:
for a hyperswap volume, can there 0 fcmaps where said volume is the target?
if so, is said volume always created with just 1 "rc_controlled" fcmap, where said volume is the source?

Reply:
What do you mean by "0 fcmaps where said volume is the target"?
No fcmap means no source/target

Reply:
actually, `fcmaps_as_target` cannot be empty and hyperswap we always have a "map" of fcmaps that point to both directions.
therefore [fcmap_as_target](https://github.com/IBM/ibm-block-csi-driver/blob/0e33b1f43d99d50ade5b5656c0f63fc80acbd2ea/controller/array_action/array_mediator_svc.py#L539) will be a non-empty list and we will [try to delete it](https://github.com/IBM/ibm-block-csi-driver/blob/0e33b1f43d99d50ade5b5656c0f63fc80acbd2ea/controller/array_action/array_mediator_svc.py#L546) and get an [exception due to fcmap that is part of remote cope](https://www.ibm.com/docs/en/flashsystem-5x00/8.1.x?topic=messages-cmmvc7154e) (i.e. `rc_controlled`). we need to add some sort of "safe delete" for `fcmap_as_target` too.

Reply:
I mean "if you count the number of fcmaps that their target is said volume, you will get 0".
members of a set can have properties, even if the set is empty (i.e. vacuously)

Reply:
no, it's impossible since each HS volume has at least one fcmap as a source and at least one fcmap as a target.

Reply:
@ArbelNathan I'm not sure I understood your "safe delete" suggestion - should we open a new conversation for it, or do you want to continue here?

Reply:
safe delete method that will prevent any rc_controlled fcmap to be deleted (or tried to be deleted..)

Reply:
Done

Reply:
so is that what we have now? can I resolve this?

Reply:
yes, its looks ok